fivegmag_ExoDvbi_player/replace.py

```diff
@@ -0,0 +1,30 @@
+import glob, os, shutil
+
+# Define paths
+src = './patch/'
+classes_dst = './ExoPlayer/demos/main/src/main/java/com/google/android/exoplayer2/demo/'
+manifest_dst = './ExoPlayer/demos/main/src/main/'
+gradle_dst = './Exoplayer/demos/main/'
+strings_dst = './Exoplayer/demos/main/src/main/res/values/'
+
+# Identify classes and copy them in the right location
+class_files = glob.iglob(os.path.join(src, "*.java"))
+for file in class_files:
+    if os.path.isfile(file):
+        shutil.copy2(file, classes_dst)
+        print("Class  " + file + "  successfully copied into Exoplayer project")
+
+# Overwrite the Android manifest
+file = './patch/AndroidManifest.xml'
+shutil.copy2(file, manifest_dst)
+print("Android manifest successfully copied into Exoplayer project")
+
+# Overwrite the build.gradle file
+file = './patch/build.gradle'
+shutil.copy2(file, gradle_dst)
+print("build.gradle successfully copied into Exoplayer project")
+
+# Overwrite the strings.xml file
+file = './patch/strings.xml'
+shutil.copy2(file, strings_dst)
+print("strings.xml successfully copied into Exoplayer project")
```

Comment:
I was wondering why you have chosen to keep whole modified files from ExoPlayer and copy them over the top rather than creating a `git diff` (or `diff -ru`) output and applying that as a patch?

In my experience, maintaining whole files means that many updates to ExoPlayer that you may wish to pull in require you to recreate some or all of these files to pull in the upstream changes, whereas a patch file (which is just the deltas) may still be able to be applied even when the upstream code has changed.

So at present to pull in upstream improvements and bug fixes you'd have to:
1. Clone the repository and the submodules
1. Run `fivegmag_ExoDvbi_player/replace.py`
1. Run `(cd fivegmag_ExoDvbi_player/ExoPlayer; git stash)`
1. Update ExoPlayer to pull in upstream changes, e.g. `(cd fivegmag_ExoDvbi_player/ExoPlayer; git checkout dev-v2)`
1. Reapply your changes: `(cd fivegmag_ExoDvbi_player/ExoPlayer; git stash apply)`
1. Fix any conflicts
1. Copy the new files out: `cp ExoPlayer/demos/main/src/main/java/com/google/android/exoplayer2/demo/{NetworkOps,PlayerActivity,WebviewActivity}.java ExoPlayer/demos/main/src/main/AndroidManifest.xml Exoplayer/demos/main/build.gradle Exoplayer/demos/main/src/main/res/values/strings.xml patch/`
1. Commit your new changed patches and the ExoPlayer submodule commit point.

With a single patch file you'd only have to update it if it no longer cleanly applied or the changes didn't work anymore. Then it would be a case of fixing the problems and running something like `(cd ExoPlayer; git add demos/main/src/main/java/com/google/android/exoplayer2/demo/WebviewActivity.java; git diff) > patch/ExoPlayer.patch` to recreate the patch file with the new changes.